lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -51,12 +51,6 @@
         description:
             - Name of the newly deployed guest
         required: True
-   name_match:
-        description:
-            - If multiple vms matching the name, use the first or last found
-        required: False
-        default: 'first'
-        choices: ['first', 'last']
```

Comment:
This parameter is only used in removed part. What do you needed for ?
IMHO two VMs with the same name is not a good practice.